### ISSUE TITLE
object_usage_linter flagging objects available during s3 method dispatch

### ISSUE THREAD

**Issue by TimTaylor** [r-lib/lintr#1768]
Are these false flags something that should / could be handled by lintr.

E.g. ideally something like the following would not trigger a warning:

``` r
lintr::lint('
Ops.myS3class <- function(e1, e2) {
    .Generic
}
'
)
#> <text>:3:5: warning: [object_usage_linter] no visible binding for global variable '.Generic'
#>     .Generic
#>     ^~~~~~~~
```
See the Technical details in `help("UseMethod")` for other similar variables that could trigger this.

**Comment by AshesITR** [comment]
Thanks for reporting this.
For convenience, here is the relevant excerpt from `help("UseMethod")`:

> `.Generic` is a length-one character vector naming the generic function.
`.Method` is a character vector (normally of length one) naming the method function. (For functions in the group generic Ops it is of length two.)
`.Class` is a character vector of classes used to find the next method. NextMethod adds an attribute "previous" to .Class giving the .Class last used for dispatch, and shifts .Class along to that used for dispatch.
`.GenericCallEnv` and `.GenericDefEnv` are the environments of the call to be generic and defining the generic respectively. (The latter is used to find methods registered for the generic.)
Note that .Class is set when the generic is called, and is unchanged if the class of the dispatching argument is changed in a method. It is possible to change the method that NextMethod would dispatch by manipulating .Class, but ‘this is not recommended unless you understand the inheritance mechanism thoroughly’ (Chambers & Hastie, 1992, p. 469).
